pkg/keystone/cronjob.go

```diff
@@ -0,0 +1,114 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keystone
+
+import (
+	keystonev1beta1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
+	common "github.com/openstack-k8s-operators/lib-common/modules/common"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/env"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// TrustFlushCommand -
+	TrustFlushCommand = "/usr/local/bin/kolla_set_configs && keystone-manage trust_flush"
+)
+
+// CronJob func
+func CronJob(
+	instance *keystonev1beta1.KeystoneAPI,
+	labels map[string]string,
+	annotations map[string]string,
+) *batchv1.CronJob {
+	runAsUser := int64(0)
+
+	args := []string{"-c"}
+	if instance.Spec.Debug.Service {
+		args = append(args, common.DebugCommand)
+	} else {
+		args = append(args, TrustFlushCommand+instance.Spec.TrustFlushArgs)
+	}
+
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_FILE"] = env.SetValue(KollaConfig)
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+
+	parallelism := int32(1)
+	completions := int32(1)
+
+	cronjob := &batchv1.CronJob{
```

Comment:
If we set `spec.successfulJobsHistoryLimit` here, we might not need to worry about adding a status condition for it? We could keep say the last n jobs?

https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#jobs-history-limits

Then a user just needs to check `oc get jobs` and the execution to see if it worked or not?

Reply:
In short, no. The concern left in keystoneapi_controller.got is about the way how we handle failure while creating/patching a CronJob resource and is not related to how actual jobs are executed. In case the operatro failed to create a cron job then there are no cronjobs and jobs created.